### PR TITLE
upgrade to go 1.19, update dependencies and simplify Dockerfile

### PR DESCRIPTION
**Pull request by bakito**
This PR updates the module to go 1.19  and some dependencies.
The go version in the Dockerfile was out of sync it the version in go. mod.
This is also updated and simplified.

Deprecated ioutil functions are corrector and some text format problems resolved.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
 LABEL stage=synobuilder
 
 RUN apk add --no-cache alpine-sdk
 WORKDIR /go/src/synok8scsiplugin
-COPY go.mod .
-RUN go mod download
-
-COPY Makefile .
-
+COPY . .
```
Comment (inductor) [inline Dockerfile:9]: This is a bad idea. You should separate COPY layer for Dockerfile best practices.

Comment (bakito) [reply Dockerfile:9]: This is only the build image which is dropped afterwards, so it died not matter.
The final image only copies the built binary from this image.

Comment (bakito) [reply Dockerfile:9]: I added a .dockerignore to skip unwanted files from the docker build

Comment (inductor) [reply Dockerfile:9]: Build cache will be used with go packages, so I beleive this should be as is

Comment (bakito) [reply Dockerfile]: Still, does not impact the final image, as the cached layers are in the builder image.
Anyway, I changed it

Comment (bakito) [reply Dockerfile]: Btw. You are using alpine:latest as source image for the final image.
You wont be able to reuse cached layers anyway, as latest is most of the time a new image. 
Also not Dockerfile best practices 🤨

**Pull request closed without merging**